### PR TITLE
Design review changes

### PR DESCRIPTION
**Pull request by alejandraarri**
Following the design review done by @urbanphes in https://github.com/CartoDB/product/issues/272#issuecomment-469661261, here is a list of the main changes:
- Markup changes to align datasets and maps condensed views
- Unify dropdowns design: filters, quick actions, search suggestions
- Add hover state in cards, tables, icons, buttons, pagination
- Update snapshot tests
- New utils file with formatting functions
- New Vue compiler option to trim whitespace

### Acceptance
Check that all the changes asked are done

### Related issues
https://github.com/CartoDB/product/issues/272
https://github.com/CartoDB/cartodb/issues/14753

### CODE REVIEW

**Comment by alejandraarri** [comment]
Retest this please

**Comment by alejandraarri** [comment]
Retest this please

**Comment by alejandraarri** [comment]
Retest this please

File: lib/assets/javascripts/new-dashboard/components/MapCard/CondensedMapCard.vue [hunk 1]
```diff
@@ -331,6 +302,10 @@ export default {
 
   .metadata__tags {
     display: flex;
+
+    li {
+      margin-right: 0.2em;
```
Comment (jesusbotella) [inline lib/assets/javascripts/new-dashboard/components/MapCard/CondensedMapCard.vue]: ver como funciona con el check de los espacios

File: lib/assets/javascripts/new-dashboard/components/Settings/Filters.vue [hunk 2]
```diff
@@ -73,20 +75,19 @@ export default {
...
-  padding: 24px 12px 24px 36px;
-  background-color: #FFF;
-}
+.filters {
+  background-color: $white;
 
-.list {
-  margin-top: 8px;
+  &-title {
+    margin-top: 20px;
```
Comment (jesusbotella) [inline lib/assets/javascripts/new-dashboard/components/Settings/Filters.vue]: Why not to use the shorthand? I don't know if there's any margin right in the additional classes.

File: webpack/new-dashboard/vue-loader.conf.js [hunk 3]
```diff
@@ -15,5 +15,8 @@ module.exports = {
     source: 'src',
     img: 'src',
     image: 'xlink:href'
+  },
+  compilerOptions: {
+    whitespace: 'condense'
```
Comment (jesusbotella) [inline webpack/new-dashboard/vue-loader.conf.js]: We need to talk about that 😂

File: lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue [hunk 4]
```diff
@@ -9,82 +9,94 @@
...
-        <span v-if="showInteractiveElements" class="card-favorite" :class="{'is-favorite': dataset.liked}" @click.prevent="toggleFavorite" @mouseover="mouseOverChildElement" @mouseleave="mouseOutChildElement">
-          <img svg-inline src="../../assets/icons/common/favorite.svg">
+
+    <div class="viz-column--main-info">
+      <div class="cell cell--start cell--first">
+        <div class="row-dataType">
+            <div class="icon--dataType" :class="`icon--${dataType}`"></div>
+        </div>
+        <span class="checkbox row-checkbox" @mouseover="mouseOverChildElement" @mouseleave="mouseOutChildElement">
+          <input class="checkbox-input" :class="{'is-selected': isSelected }" @click.stop.prevent="toggleSelection($event)" type="checkbox">
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue]: Done!

File: lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue [hunk 5]
```diff
@@ -9,82 +9,94 @@
...
-        :dataset="dataset"
-        :isShared="isShared"
-        class="dataset--quick-actions"
-        @open="openQuickActions"
-        @close="closeQuickActions"
-        @contentChanged="onContentChanged"/>
+      <div class="viz-column--share">
+        <div class="cell cell--small">
+          <span class="text is-small is-txtSoftGrey" v-if="!dataset.dependent_visualizations_count">
+            {{ $tc(`DatasetCard.maps`, 0, { n: 0 }) }}
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue]: Done!

File: lib/assets/javascripts/new-dashboard/components/Settings/Filters.vue (hunk 2 above)
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Settings/Filters.vue]: Actually, it should be a padding.

File: lib/assets/javascripts/new-dashboard/components/MapCard/CondensedMapCard.vue (hunk 1 above)
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/MapCard/CondensedMapCard.vue]: It works!!

File: lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue [hunk 6]
```diff
@@ -9,82 +9,94 @@
...
-          <span class="text is-small is-txtSoftGrey dataset__dependent-visualizations">
-            {{ $tc(`DatasetCard.maps`, dataset.dependent_visualizations_count, { n: dataset.dependent_visualizations_count }) }}
+
+    <div class="viz-column--extra-info">
+      <div class="viz-column--status">
+        <div class="cell cell--large">
+          <span class="text is-small is-txtSoftGrey">{{ lastUpdated }}</span>
+        </div>
+        <div class="cell cell--small">
+          <span class="text is-small is-txtSoftGrey" :title="$tc(`DatasetCard.numberRows`, dataset.table.row_count, { count: getNumberInLocaleFormat(dataset.table.row_count) })">
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue]: Done!

File: lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue [hunk 7]
```diff
@@ -9,82 +9,94 @@
...
+            {{ $tc(`DatasetCard.numberRows`, dataset.table.row_count, { count: numberFormatter(dataset.table.row_count) }) }}
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/new-dashboard/components/Dataset/DatasetCard.vue]: Done!

**Pull request merged**